### DISCUSSION TITLE
How can I add logging to a Rocket app?

### DISCUSSION THREAD

**Discussion by szabgab** [rwf2/Rocket#2704] [Questions] [answered]
It seems that I can call `rocket::log::info_!("blabla");` and it will be printed to the console along with the rest of the logs, but that trailing underscore does not give me confidence.

I could not find anything relevant in the docs, but my search fu is not very high.

Is there an official way to log from a Rocket app?

**Comment by ELD** [comment, marked as answer]
Those macros are actually reexported without the underscore, so you can do the same thing with just `rocket::info!("blah")` rather than fully pathing the module.

That said, there's a longstanding issue to improve logging: https://github.com/rwf2/Rocket/issues/21

There have been a couple attempts to replace the current logging infrastructure with tracing to allow for more flexibility, but it's unfortunately never been merged, yet. I'd guess we get improved logging sometime in 0.7 or 0.8!

**Comment by szabgab** [comment]
Ah, without the `log` part. I see. Thanks. It would be still nice to add this to the [Guide](https://rocket.rs/v0.5/guide/) before 0.7.

**Comment by szabgab** [comment]
In any case I [posted about logging](https://rust.code-maven.com/rocket-logging).
